app/services/form_to_resource_service.rb

```diff
@@ -113,5 +114,20 @@ def new_contributor(given_name, family_name, orcid, type, sequence)
         return if family_name.blank? && given_name.blank? && orcid.blank?
         PDCMetadata::Creator.new_contributor(given_name, family_name, orcid, type, sequence)
       end
+
+      # Funders:
+
+      def process_funders(params, resource)
+        # (New pattern: Use rails param name conventions rather than numbering fields.)
+        resource.funders = (params[:funders] || []).filter_map do |funder|
+          new_funder(funder[:funder_name], funder[:award_number], funder[:award_uri])
+        end
+        # (New pattern: no return -- method modifies resource in place.)
+      end
+
+      def new_funder(funder_name, award_number, award_uri)
+        return if funder_name.blank? && award_number.blank? && award_uri.blank?
```

Comment:
This explains why the need for filter map...

Reply:
Right: This is pattern that is already used for creator and contributor, but with an extra `if` at the end. If there's an empty row in the table, we don't create a record for it.